### PR TITLE
Auto-unmount repos in pre-install

### LINKED ISSUE #1124: Installer should unmount first before installing
`preflight` script in the Mac installer should unmount repositories before proceeding with the install.

### PR DESCRIPTION
**Pull request by alameenshah**
Mac installer pkg would now automatically unmount all repositories registered for auto mount

Fixes #1124

### CODE REVIEW

File: GVFS/GVFS.Installer.Mac/scripts/preinstall [hunk 1]
```diff
@@ -17,3 +17,10 @@ if [ "$isKextLoaded" -gt 0 ]; then
     echo $unloadCmd
     eval $unloadCmd || exit 1
 fi
+
+GVFSBINPATH="/usr/local/vfsforgit/gvfs"
+if [ -f "${GVFSBINPATH}" ]; then
+    unmountCmd="${GVFSBINPATH} service --unmount-all"
+    echo $unmountCmd
+    eval $unmountCmd || exit 1
```
Comment (jeschu1) [inline GVFS/GVFS.Installer.Mac/scripts/preinstall]: Just to double check will the eval print any errors that come up from the call?  
This follows the format of the other commands and lgtme.

Comment (alameenshah) [reply GVFS/GVFS.Installer.Mac/scripts/preinstall]: yes it does.

File: GVFS/GVFS.Installer.Mac/scripts/Distribution.xml [hunk 2]
```diff
@@ -34,7 +34,7 @@
         
         function GetBlockingProcesses()
```
Comment (wilbaker) [inline GVFS/GVFS.Installer.Mac/scripts/Distribution.xml]: Does this get called before or after `preinstall`?

Comment (alameenshah) [reply GVFS/GVFS.Installer.Mac/scripts/Distribution.xml]: Before `preinstall`. 

This is the order - `Distribution.xml::InstallationCheck`, `preinstall`, `postinstall`.

**Comment by Ignas** [comment]
I think this will fail if you are running installer as root with a clean env (like using Munki or just sudo install) and should rather be:
```
for username in $(ps -Ac -o user,command | grep -iw "loginwindow" | awk '{print $1}'); do
    USER_HOME=$(eval echo "~$username")
    HOME="${USER_HOME}" gvfs service --unmount-all
done
```

Same as the service restart commands.

**Pull request merged**